pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+
+        Returns:
+
+            : The returned list of objects depend on the usage:
+
+            * For instance-level use: A list of
+              :class:`~pywbem.CIMInstanceName` objects that are the instance
+              paths of the referencing association instances, with their
+              attributes set as follows:
+
+              * `classname`: Name of the creation class of the instance.
+              * `keybindings`: Keybindings of the instance.
+              * `namespace`: Name of the CIM namespace containing the instance.
+              * `host`: Host and optionally port of the WBEM server containing
+                the CIM namespace, or `None` if the server did not return host
+                information.
+
+            * For class-level use: A list of :class:`~pywbem.CIMClassName`
+              objects that are the class paths of the referencing association
+              classes, with their attributes set as follows:
+
+              * `classname`: Name of the class.
+              * `namespace`: Name of the CIM namespace containing the class.
+              * `host`: Host and optionally port of the WBEM server containing
+                the CIM namespace, or `None` if the server did not return host
+                information.
+
+        Raises:
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_
+
+
+        """
+
+        if isinstance(ObjectName, CIMInstanceName):
+            ref_paths = self._get_reference_instnames(namespace, ObjectName,
+                                                      ResultClass,
+                                                      Role)
+            rtn_names = [r.copy() for r in ref_paths]
+
+            for iname in rtn_names:
+                if iname.host is None:
+                    iname.host = self.host
+
+            return rtn_names
+
+        ref_classnames = self._get_reference_classnames(namespace, ObjectName,
+                                                        ResultClass, Role)
+
+        ref_result = [CIMClassName(classname=cn, host=self.host,
+                                   namespace=namespace)
+                      for cn in ref_classnames]
+
+        return ref_result
+
+    ####################################################################
+    #
+    #   References server request
+    #
+    ####################################################################
+
+    def References(self, namespace, ObjectName, ResultClass=None, Role=None,
+                   IncludeQualifiers=None, IncludeClassOrigin=None,
+                   PropertyList=None):
+        # pylint: disable=line-too-long
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.References`
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
DONE